packages/jest-runtime/src/index.ts

```diff
@@ -2231,8 +2231,36 @@ export default class Runtime {
     const jestObject: Jest = {
       advanceTimersByTime: (msToRun: number) =>
         _getFakeTimers().advanceTimersByTime(msToRun),
+      advanceTimersByTimeAsync: async (msToRun: number): Promise<void> => {
+        const fakeTimers = _getFakeTimers();
+
+        if (
+          fakeTimers === this._environment.fakeTimersModern &&
+          typeof fakeTimers.advanceTimersByTimeAsync === 'function'
```

Comment:
This and other similar checks are just temporary, right?

```suggestion
          // TODO: remove this check in Jest 30
          typeof fakeTimers.advanceTimersByTimeAsync === 'function'
```

Reply:
yep

Reply:
should probably be

```ts
if (fakeTimers === this._environment.fakeTimersModern) {
  // TODO: remove this check in Jest 30
  if (typeof fakeTimers.advanceTimersByTimeAsync !== 'function') {
    throw new TypeError(
      'Your test environment does not support async fake timers - please ensure its Jest dependencies are updated to version 29.5 or later',
    );
  }
  await fakeTimers.advanceTimersByTimeAsync(msToRun);
} else {
  throw new TypeError(
    '`jest.advanceTimersByTimeAsync()` is not available when using legacy fake timers.',
  );
}
```